modules/aidemBidAdapter.js

```diff
@@ -0,0 +1,691 @@
+    deepSetValue(errorNotice.metrics, 'adapter.client.interpretResponse', metrics['adapter.client.interpretResponse'])
+  }
+  return errorNotice
+}
+
+/**
+ * Get One size from Size Array
+ * [[250,350]] -> [250, 350]
+ * [250, 350]  -> [250, 350]
+ * @param {array} sizes array of sizes
+ */
+function getFirstSize(sizes = []) {
+  if (isValidSize(sizes)) {
+    return sizes;
+  } else if (isValidSize(sizes[0])) {
+    return sizes[0];
+  }
+
+  return false;
+}
+
+function hasValidFloor(obj) {
+  if (!obj) return false
+  const hasValue = !isNaN(Number(obj.value))
+  const hasCurrency = contains(AVAILABLE_CURRENCIES, obj.currency)
+  return hasValue && hasCurrency
+}
+
+function getMediaType(bidRequest) {
+  if ((bidRequest.mediaTypes && bidRequest.mediaTypes.hasOwnProperty('video')) || bidRequest.params.hasOwnProperty('video')) { return VIDEO }
+  return BANNER
+}
+
+function getPrebidRequestFields(bidderRequest, bidRequests) {
+  const payload = {}
+  // Base Payload Data
+  deepSetValue(payload, 'id', bidderRequest.auctionId);
+  // Impressions
+  setPrebidImpressionObject(bidRequests, payload)
+  // Device
+  deepSetValue(payload, 'device', getDevice())
+  // Timeout
+  deepSetValue(payload, 'tmax', bidderRequest.timeout);
+  // Currency
+  deepSetValue(payload, 'cur', DEFAULT_CURRENCY);
+  // Timezone
+  deepSetValue(payload, 'tz', new Date().getTimezoneOffset());
+  // Privacy Regs
+  deepSetValue(payload, 'regs', getRegs());
+  // Site
+  setPrebidSiteObject(bidderRequest, payload)
+  // Environment
+  setPrebidRequestEnvironment(payload)
+  // AT auction type
+  deepSetValue(payload, 'at', 1);
+  // User
+  payload.user = {};
+
+  return payload
+}
+
+function setPrebidImpressionObject(bidRequests, payload) {
+  payload.imp = [];
+  payload.mediaTypes = []
+  _each(bidRequests, function (bidRequest) {
+    const impressionObject = {};
+    // Placement or ad tag used to initiate the auction
+    deepSetValue(impressionObject, 'id', bidRequest.bidId);
+    // Site id
+    deepSetValue(impressionObject, 'siteId', deepAccess(bidRequest, 'params.siteId'));
+    // Tag id
+    deepSetValue(impressionObject, 'tid', deepAccess(bidRequest, 'transactionId'));
+    const mediaType = getMediaType(bidRequest)
+    if (!payload.mediaTypes.includes(mediaType)) {
+      payload.mediaTypes.push(mediaType)
+    }
+    switch (mediaType) {
+      case 'banner':
+        setPrebidImpressionObjectBanner(bidRequest, impressionObject)
+        break;
+      case 'video':
+        setPrebidImpressionObjectVideo(bidRequest, impressionObject)
+        break;
+    }
+
+    // Floor (optional)
+    setPrebidImpressionObjectFloor(bidRequest, impressionObject)
+
+    impressionObject.imp_ext = {};
+
+    payload.imp.push(impressionObject);
+  });
+}
+
+function setPrebidSiteObject(bidderRequest, payload) {
+  deepSetValue(payload, 'site.domain', deepAccess(bidderRequest, 'refererInfo.domain'));
```

Comment:
@dgirardi perhaps you can offer some advise here? There's a much more straightforward way to they've much of this